### PR TITLE
Implement `add` and `addu` instructions

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR implements the `add` and `addu` instructions. This builds upon https://github.com/o1-labs/proof-systems/pull/1397.

### CODE REVIEW

File: optimism/src/mips/interpreter.rs [hunk 1]
```diff
@@ -545,8 +545,22 @@ pub fn interpret_rtype<Env: InterpreterEnv>(env: &mut Env, instr: RTypeInstructi
         RTypeInstruction::MultiplyUnsigned => (),
         RTypeInstruction::Div => (),
         RTypeInstruction::DivUnsigned => (),
-        RTypeInstruction::Add => (),
-        RTypeInstruction::AddUnsigned => (),
+        RTypeInstruction::Add => {
```
Comment (dannywillems) [inline optimism/src/mips/interpreter.rs:548]: May cause overflow exception

Comment (dannywillems) [reply optimism/src/mips/interpreter.rs:548]: Do we handle it?

Comment (mrmr1993) [reply optimism/src/mips/interpreter.rs:548]: Unimplemented in cannon :man_shrugging:

Comment (dannywillems) [reply optimism/src/mips/interpreter.rs:548]: Oh, interesting.

Comment (dannywillems) [reply optimism/src/mips/interpreter.rs:548]: Opening a follow-up.

**Review by dannywillems — APPROVED** [review]
Accepting, opened follow-ups.

**Pull request merged**